diff --git a/meson.build b/meson.build
index 7e816d8b29..0ac66d3e98 100644
--- a/meson.build
+++ b/meson.build
@@ -26,6 +26,9 @@
 libswresample = dependency('libswresample', version: '>= 4.12.100')
 libswscale = dependency('libswscale', version: '>= 7.5.100')
 
+# fftools-ffi
+libfftools_ffi = dependency('fftools-ffi')
+
 libplacebo = dependency('libplacebo', version: '>=6.338.2',
                 default_options: ['default_library=static', 'demos=false'])
 
@@ -40,7 +43,8 @@
                 libavutil,
                 libplacebo,
                 libswresample,
-                libswscale]
+                libswscale,
+                libfftools_ffi]
 
 # Keeps track of all enabled/disabled features
 features = {
@@ -265,6 +269,9 @@
     'ta/ta.c',
     'ta/ta_talloc.c',
     'ta/ta_utils.c',
+
+    ## fftools-ffi hack
+    'fftools-ffi.c'
 )
 
 
